bagel/tests/test_cli_bids.py

```diff
@@ -151,3 +156,39 @@ def test_create_acquisitions(bids_path, bids_dir, acquisitions, bids_session):
 
     for contrast, count in acquisitions.items():
         assert image_counts[contrast] == count
+
+
+@pytest.mark.parametrize(
+    "bids_sub_id, session",
+    [("01", "01"), ("02", "02"), ("03", "01")],
+)
+def test_get_session_path_session_exists(bids_sub_id, session):
+    """Given a subject and session ID, get_session_path() returns a path to the subject's session directory."""
+    bids_dir = Path(__file__).parent / "../../bids-examples/synthetic"
+    session_path = get_session_path(
+        layout=BIDSLayout(bids_dir, validate=True),
+        bids_dir=bids_dir,
+        bids_sub_id=bids_sub_id,
+        session=session,
+    )
+
+    assert f"sub-{bids_sub_id}" in session_path
+    assert f"ses-{session}" in session_path
+    assert Path(session_path).is_absolute()
+    assert Path(session_path).is_dir()
+
+
+@pytest.mark.parametrize("bids_sub_id", ["01", "03", "05"])
+def test_get_session_path_session_missing(bids_sub_id):
+    """Given only a subject ID, get_session_path() returns the path to the BIDS subject directory."""
```

Comment:
the test name is pretty clear already, but I'd also add to the docstring, that this is testing a dataset where the session layer is missing.